api/hooks/validate/validators/dates.validator.js

```diff
@@ -0,0 +1,44 @@
+const isValidDateFormat = string => {
```

Comment:
These all look good to me, and no objection to merging. Wondering though if there's a library we can pull in that does these validations though.

Reply:
The date validations were spotty in the old validation library - they didn't validate for actual dates (ie, 2019-02-30 and 2019-09 were both considered valid), so we created these validators. I didn't deeply test the date validators in the new library, but in my initial tests there were similar issues, so just decided to bring our custom validators over.